### ISSUE TITLE
Connecting to Raspberry Pi

### ISSUE THREAD

**Issue by Tissy** [alphafox02/antsdr_dji_droneid#2]
Hi,

Do you have any experience with connecting the SDR to a RaspberryPi please?

I am using a USB RJ45 dongle which I have fixed the IP to 192.168.1.2, however I cannot ping the ANTSDR on IP 192.168.1.10.

Any suggestions please as I cannot find any suitable resources to assist.

Thank you for your help.

**Comment by alphafox02** [comment]
Use the PIs native Ethernet port vs the external one. Are you using the firmware here specific to the dji piece?

**Comment by Tissy** [comment]
Hi, thank you very much for your reply, much appreciated.  I've been trying to resolve this for 2 days now, so your help is appreciated.

I have tried connecting it to the main CAT5 port, but then I loose remote access (SSH) to the Pi if I am connecting via the external adapter.  Do you perhaps have any details of how the IP addresses should be configured and whether or not to use DHCP or fixed IPs.

I currently have:
```
eth0 fixed at 192.168.1.108 (RPI RJ45)
eth1 fixed at 192.168.1.11 (USB RJ45 Dongle)
wlan fixed at 192.168.1.109
```

Thank you again,

Steve

**Comment by alphafox02** [comment]
Depending on your network setup, I’d use eth0 as the connection to the AntSDR and use the usb for Ethernet and assign it static, but something like 192.168.2.100 and then your laptop 192.168.2.101. I’m not sure what you’re using the WiFi interface for, is it setting up an access point for you to connect to? If so, I’d put it on a different range. You should then be able to ssh the pi at 192.168.2.101 and the pi in turn would be able to communicate with the AntSDR as needed.

**Comment by Tissy** [comment]
That's really helpful, thank you.  I feel I am getting somewhere with this.

Is there a reason that the Ethernet adapter (USB - RJ45) needs to be on a different subnet to the main Pi RJ45, which will be on 192.168.1.X.

Changing the Ethernet adapters fixed IP to the 192.168.2.X subnet is preventing outgoing traffic to the TAK server (as my gateway is on 192.168.1.254), so trying to figure out best options.

Thank you once again.

**Comment by alphafox02** [comment]
Might be best to eliminate the USB adapter all together and plug the AntSDR and Pi Eth port into your network together. Or if you have a switch, that’d work too. Then you can put everything on the same network since you happen to have the same IP scheme. I’m not totally sure you’d have to change the IPs as I suggested, it just had me thinking it may be an issue to have them all setup like that.

**Comment by vasmane** [comment]
U have to disable ipv6 and set ip on ipv4, that resolves the problem on both rpi and dragon os. Sometimes with different adapters i cannot ping ant e200, depends on the adapter

**Comment by Tissy** [comment]
Hi @vasmane, thank you for your comment, much appreciated.  It's very frustrating trying to figure out what is causing the issue.

I have tried two different USB-RJ45 adaptors and unable to ping the AntSDR if that is plugged into the adapter, but can access the Pi via its normal RJ45 port.

If I swap over and connect the AntSDR to the main RJ45 connector and the LAN to the adapter, I cannot access the Pi.

Using a LAN viewer, I can see each of the ports on the Pi, but they all appear to have the same MAC address which is weird. (see attached)
 
<img width="652" alt="image" src="https://github.com/alphafox02/antsdr_dji_droneid/assets/1116396/37adbdab-1341-47e5-9ae9-21ad92b2b8a2">

This is what `ifconfig` shows.
`eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.1.2  netmask 255.255.255.0  broadcast 192.168.1.255
        ether dc:a6:32:4b:49:4e  txqueuelen 1000  (Ethernet)
        RX packets 24187  bytes 1709680 (1.6 MiB)
        RX errors 0  dropped 2  overruns 0  frame 0
        TX packets 1305  bytes 191096 (186.6 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.1.11  netmask 255.255.255.0  broadcast 192.168.1.255
        ether a0:ce:c8:35:f2:cb  txqueuelen 1000  (Ethernet)
        RX packets 2763  bytes 162206 (158.4 KiB)
        RX errors 0  dropped 2  overruns 0  frame 0
        TX packets 46  bytes 5295 (5.1 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 174  bytes 16571 (16.1 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 174  bytes 16571 (16.1 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.1.24  netmask 255.255.255.0  broadcast 192.168.1.255
        inet6 fe80::382e:27e7:7b9f:f891  prefixlen 64  scopeid 0x20<link>
        inet6 2a0a:ef40:c7d:ab01:3d43:d821:6d80:6286  prefixlen 64  scopeid 0x0<global>
        ether dc:a6:32:4b:49:4f  txqueuelen 1000  (Ethernet)
        RX packets 1413  bytes 273419 (267.0 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 153  bytes 18154 (17.7 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
`
Any suggestions please?

**Comment by vasmane** [comment]
What kind of power supply are you using? The rpi 5 needs 25 wats, otherwise it goes into low power mode, that is probably causing the problem. Im using a usb a dongle with rj45 connector and the original 5v 5a power supply with no issues whatsoever. Its mandatory that you disable ipv6 to the adapter, otherwise it will disconnect every minute or so.

**Comment by Tissy** [comment]
Thank you.  I am using a proper Pi 5v 3A (15.3w) PSU and it's actually a Pi4 I am using.

Looking at ifconfig, I have disabled IPv6 using the `sudo nano /etc/sysctl.conf` and then adding `net.ipv6.conf.all.disable_ipv6 = 1` to the file.

**Comment by Tissy** [comment]
Which USB adapters have you used and know work please @vasmane

**Comment by alphafox02** [comment]
I can check with a Pi4, AntSDR, and adapter with the same setup after I get back in a week or so. Hopefully you get it beforehand though. For a temp fix though, toss a gig switch in the mix. Cable into switch, out to both AntSDR and Pi4 stock port. Should get you to the AntSDR and your 192.168.x.x network at the same time.

**Comment by OpenCV-Peter** [comment]
I do not think it is a good idea to use the same subnet (here 192.168.1.0/24) on multiple network ports. That might mess up the IP routing.
